### ISSUE TITLE
Remove the `join__owner` directive.

### ISSUE THREAD

**Issue by pcmanus** [apollographql/specs-join#16]
In the relaxed composition model, we don't have type ownership anymore (we will ultimately adds governance options that optionally restore ownership if ones want, but this will be in the form of additional composition validation), so `join__owner` won't be used and it's probably cleaner/simpler to remove it (even if we preserve preserving it as optional, which I'd discourage but is technically possible, the spec needs to be adapted accordingly)?.

**Comment by queerviolet** [comment]
it strikes me that if we retain information about whether a subgraph contains a definition of a type, we could implement ownership—even on a per-type basis—as a supergraph validation. would we like to do that?

**Comment by queerviolet** [comment]
the proposal would be something like `@join__type(extension: Boolean = false)`

**Comment by pcmanus** [comment]
> we could implement ownership—even on a per-type basis—as a supergraph validation. would we like to do that?

Afaict, in the short term (meaning fed 2 GA horizon), the only place that will care about ownership is going to be the legacy mode, and since we decided this should generate existing (join spec v0.1) supergraphs, it will have to be done pre-supergraph generation.

*That said*, I'm actually in favour of that proposal: it's trivial to do and I'm sure we'll be happy to have that information available at some point. Happy to add such `extension` argument to `@join__type` on top of removing `join__owner` here.

**Comment by queerviolet** [comment]
yeah, i don't really mean ownership per se, but the semantics of "this type is defined once, and keys and requires can only be from that definition". or perhaps something similar but not quite the same, depending on actual governance needs

**Comment by jeffjakub** [comment]
@pcmanus - Is this issue still needed?

**Comment by pcmanus** [comment]
Depends a bit.

The `@join__owner` directive is not part of `join/v0.2` and https://specs.apollo.dev/join/v0.2/ does not "list it" in its section 6 "Directives". That said, the examples of that doc could use some update since they are still referring to it, and are thus not completely up-to-date with the doc content. Overall, https://specs.apollo.dev/join/v0.2/ could probably use a bit of cleanup since it also doesn't mention `@join__implements` even though it is technically part of `join/v0.2`.

But the overall context is that IMO, the ROI of massaging https://specs.apollo.dev/join/v0.2/ is currently pretty low: the "join spec" is largely an internal implementation detail of the supergraph and I don't see the point of spending time on a precise spec. So I'd suggest just timeboxing 1h of someones time to fixup the examples at https://specs.apollo.dev/join/v0.2/ (maybe just by removing them) in general and ensure the definitions listed are up-to-date, but I'd leave it at that for that spec for the short/mid-term. That's my opinion though and others may have different ones.
